### DISCUSSION TITLE
NaN Values

### DISCUSSION THREAD

**Discussion by Easiii** [mindsphere/node-red-contrib-mindconnect#158] [Q&A] [answered]
Hello,

I'm using the MindConnect Node to handle different types of data eg. temperature values.
If a Sensor does not work properly he might provide NaN Values.

Is there a way to handle NaN values with the MindConnect Node or the MindConnect function node?

I could sort out all NaN values ​​as they arise, but I'm looking for a central solution at the end of the process.

KR
Easiii

**Comment by sn0wcat** [comment, marked as answer]
You can write a small javascript method in the function node which is preparing the data for mindconnect node which filters out the NaN values before you send the data to mindsphere node.

**Reply by Easiii** [reply]
Okay thats sounds like what I'm looking for.
Any ideas how to write such a script?

**Reply by peterandree** [reply]
Something like this

```javascript
msg.payload = msg.payload.filter(function (e) {
    return !isNaN(e.value) ;
});
```

**Reply by Easiii** [reply]
Yes thank you! 
I did some testing and it works as intendet!
